BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Services/Users/UserService.cs

```
@@ -0,0 +1,37 @@
+﻿using BlazorMarkDownAppJwt.Server.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace BlazorMarkDownAppJwt.Server.Services.Users
+{
+    public class UserService : IUserService
+    {
+        private DataBaseContext Ctx { get; set; }
+
+        public UserService(DataBaseContext ctx)
+        {
+            Ctx = ctx;
+        }
+
+        public async Task<User?> AddUser(User addedUser)
+        {
+            if (string.IsNullOrWhiteSpace(addedUser.Email)
+                || string.IsNullOrWhiteSpace(addedUser.Password)
+                || string.IsNullOrWhiteSpace(addedUser.FirstName)
+                || string.IsNullOrWhiteSpace(addedUser.LastName))
+                return null;
+
+            var existingUser = await Ctx.Users.SingleOrDefaultAsync(u => u.Email == addedUser.Email);
+            if (existingUser != null) return null;

```

You should throw an exception in order to manage user exists